internal/amss/service.go

```diff
@@ -0,0 +1,11 @@
+package amss
+
+import (
+	"context"
+	"io"
+)
+
+type Service interface {
```

Comment:
Why calling it `Service`? And then `Client` and `NewClient`? We have called them `Client` and `clientImpl` in other interfaces/implementations.

Reply:
I've gone with the name "pooledClient" for your AMSS client implementation because it's based on the hashicorp/cleanhttp pooled HTTP client.